### PR TITLE
Remove reasons from iptables syncProxyRules

### PR DESCRIPTION
**Pull request by wojtek-t**
The reasons are no longer useful, since we know if something changed anyway, I think.

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot kops test this

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -1495,8 +1481,8 @@ func (proxier *Proxier) syncProxyRules(reason syncReason) {
 	}
 	proxier.portsMap = replacementPortsMap
 
-	// Update healthz timestamp if it is periodic sync.
-	if proxier.healthzServer != nil && reason == syncReasonForce {
```
Comment (MrHohn) [inline pkg/proxy/iptables/proxier.go]: We check sync reason here to make sure we only update healthz timestamp in periodic sync. Deleting it means we now update timestamp on every sync, which may increase lock contention with the healthz server. Would it increase the risk that DDOSing the healthz server will halt syncProxyRules thread?

cc @nicksardo

Comment (MrHohn) [inline pkg/proxy/iptables/proxier.go]: Ah I think it should be fine, doesn't seem like [Load()](https://golang.org/src/sync/atomic/value.go?s=798:836#L21) will block [Store()](https://golang.org/src/sync/atomic/value.go?s=1308:1344#L38):
```
// Load returns the value set by the most recent Store.
```

**Pull request merged**